### PR TITLE
trinity: Version Update to 2.12.0

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

**Comment by dorton21** [comment]
Depends on #23457

**Comment by dorton21** [comment]
Is this failing simply due to "Depends on #23457"?

File: var/spack/repos/builtin/packages/trinity/package.py [hunk 1]
```diff
@@ -36,6 +38,14 @@ class Trinity(MakefilePackage):
...
     #  and we have runtime errors without them
     # https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Docker/Dockerfile
+    depends_on("r-dexseq", type="run")
+    depends_on("star", type="run")
+    depends_on("picard", type="run")
+    depends_on("subread", type="run")
+    depends_on("gatk", type="run")
+    depends_on("gmap-gsnap", type="run")
+    depends_on("r-tximport", type="run")
+    depends_on("r-tximportdata", type="run")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/trinity/package.py]: Do these dependencies apply to `2.6.6`?  If not, then a `when` argument should be provided (e.g., `when='@2.12')`).

File: var/spack/repos/builtin/packages/trinity/package.py [hunk 2]
```diff
@@ -20,10 +20,12 @@ class Trinity(MakefilePackage):
     """
 
     homepage = "http://trinityrnaseq.github.io/"
-    url      = "https://github.com/trinityrnaseq/trinityrnaseq/archive/Trinity-v2.6.6.tar.gz"
+    url      = "https://github.com/trinityrnaseq/trinityrnaseq/releases/download/v2.12.0/trinityrnaseq-v2.12.0.FULL.tar.gz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/trinity/package.py]: This change breaks the `2.6.6` build.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/trinity/package.py]: Given the differences in urls between versions, you may want to consider either the `url_for_version` method or adding the `url` argument to the `version` directive.

See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=version%20url%3D#version-urls

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by dorton21** [comment]
#23457 Has been merged

**Comment by dorton21** [comment]
Which requirement is not being met for auto-merge?

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
> Which requirement is not being met for auto-merge?

An approval.

**Pull request merged**